lib/ufe/UsdStageMap.cpp

```diff
@@ -15,6 +15,43 @@
 //
 
 #include "UsdStageMap.h"
+#include "Utils.h"
+
+#include <maya/MFnDagNode.h>
+
+#include <cassert>
+
+namespace {
+
+MObjectHandle proxyShapeHandle(const Ufe::Path& path)
+{
+    // Get the MObjectHandle from the tail of the MDagPath.  Remove the leading
+    // '|world' component.
+    auto noWorld = path.popHead().string();
+    auto dagPath = MayaUsd::ufe::nameToDagPath(noWorld);
+    MObjectHandle handle(dagPath.node());
+    if (!handle.isValid()) {
+        TF_CODING_ERROR("'%s' is not a path to a proxy shape node.",
+                        noWorld.c_str());
+    }
+    return handle;
+}
+
+// Assuming proxy shape nodes cannot be instanced, simply return the first path.
```

Comment:
I'll admit I'm not sure I fully understand the context here, but is this a safe assumption? In our pipeline, we definitely do instance `pxrUsdReferenceAssembly`/`pxrUsdProxyShape` nodes using either nParticles or MASH.

Reply:
For some additional context, the scene file for the `testPxrUsdMayaGLInstancerDraw` unit test is an example of an instancer setup:
https://github.com/Autodesk/maya-usd/blob/dev/plugin/pxr/maya/lib/pxrUsdMayaGL/testenv/InstancerDrawTest/InstancerDrawTest.ma
https://github.com/Autodesk/maya-usd/blob/dev/plugin/pxr/maya/lib/pxrUsdMayaGL/testenv/testPxrUsdMayaGLInstancerDraw.py

Reply:
Created an issue describing the problem, and some areas that would need to be worked on:
https://github.com/Autodesk/maya-usd/issues/256